spec/factories/courses.rb

```diff
@@ -9,90 +9,90 @@
       Course.find_by(identifier:) || new(**attributes)
     end
 
-    trait :ehco do
+    trait :early_headship_coaching_offer do
       sequence(:name) { |n| "NPQ Early Headship Coaching Offer Course #{n}" }
       identifier { "npq-early-headship-coaching-offer" }
-      course_group { CourseGroup.find_by(name: "ehco") || create(:course_group, name: "ehco") }
+      course_group { CourseGroup.find_or_create_by(name: "ehco") }
```

Comment:
As we're using `initialize_with` for a lot of the factories now, can we also use it for `CourseGroup`, that way we can just use `create(:course_group, name: "ehco")` here and the other traits below.